knowledge-base/grid-allow-prevent-row-selection.md

```diff
@@ -0,0 +1,109 @@
+---
+title: Allow/Prevent Selection of Certain Row in the Grid
+description: Learn how to manage row selection in the Telerik Blazor Grid by preventing specific rows from being selected conditionally, applying CSS rules to checkbox columns, and handling selection changes through the SelectedItemsChanged event.
+type: how-to
+page_title: Allow/Prevent Selection of Certain Row in the Grid
+slug: grid-kb-prevent-row-selection
+tags: grid, selection
+ticketid: 1441291
+res_type: kb
+---
+
+## Environment
+
+<table>
+    <tbody>
+        <tr>
+            <td>Product</td>
+            <td>Grid for Blazor</td>
+        </tr>
+    </tbody>
+</table>
+
+## Description
+
+This KB article answers the following questions:
+
+* How to disable the selection of a row in the Grid?
+* How to prevent the selection of a certian row in the Grid?
+* How to disable the checkbox column of a disabled row in a Telerik Blazor Grid?
+* How to customize the checkbox column of a certain row?
+* How to prevent the selection of Grid items conditionally through the [SelectedItemsChanged]({%slug grid-events%}#selecteditemschanged) event?
+
+## Solution
+
+The key points in the required implementation are:
+
+* Use the [OnRowRender]({%slug grid-events%}#onrowrender) handler to set the custom class to the desired row. 
+* Use the `SelectedItemsChanged` event of the Grid to prevent the selection of items conditionally.
+* Disable the checkboxes by conditionally applying classes to the rows in which the checkboxes should be disabled.
```

Comment:
Avoid [using ambiguous modal verbs](https://docs.telerik.com/style-guide/basic-rules-and-guidelines#use-properly-should-could-must-have-to-need-to) like "should".
```suggestion
* Disable the checkboxes by conditionally applying classes to the rows where the checkboxes will be disabled.
```